### ISSUE TITLE
Error: 'Name': 'That URL is already in use.' 

### ISSUE THREAD

**Issue by etj** [geosolutions-it/ckanext-dcatapit#168]
```
{'Name': 'That URL is already in use.'}
```

See also #155: this error is triggered at least in 
- Regione Toscana
- Comune di Milano
- Regione Piemonte

**Comment by etj** [comment]
Example:

The local content for a dataset which was not imported due to this error, had:
- `title`: "Strutture territoriali private di assistenza psichiatrica e di patologia da dipendenza (SERT) - per tipologia **struttura**",
- `name`: "strutture-territoriali-private-di-assistenza-psichiatrica-e-di-patologia-da-dipendenza-sert-per",

WIthin the local CKAN, that name reported a dataset with
- `title`: "Strutture territoriali private di assistenza psichiatrica e di patologia da dipendenza (SERT) - per tipologia **Unita' Funzionale**"

**Comment by etj** [comment]
ckanext-dcat/harvest/base.py
```python
        if not package_dict.get('name'):
            package_dict['name'] = self._get_package_name(harvest_object, package_dict['title'])
```
then
```python
    def _get_package_name(self, harvest_object, title):

        package = harvest_object.package
        if package is None or package.title != title:
            name = self._gen_new_name(title)
            if not name:
                raise Exception('Could not generate a unique name from the title or the GUID. Please choose a more unique title.')
        else:
            name = package.name

        return name
```
Method `_gen_new_name` ensure no dup name is used, but it will never be called since `package.title == title`.

Proably the line 
```python
package_dict['name'] = self._get_package_name(harvest_object, package_dict['title'])
```
Should be changed into 
```python
package_dict['name'] = self._get_package_name(harvest_object, None)
```

**Comment by cezio** [comment]
Fix in #172 will enforce name generation for each newly created dataset based on a title.
